test-count.js

```diff
@@ -0,0 +1,327 @@
+    jsonContent,
+    { encoding: "utf8" },
+    (err) => {
+      if (err) {
+        throw err;
+      }
+    }
+  );
+}
+
+function testCountTemplate(testCountMap, packageName, testTag) {
+  testCountMap.set(
+    `${packageName}_${testTag}`,
+    testCountMap.get(`${packageName}_${testTag}`) + 1 || 1
+  );
+}
+
+function lineParser(lineText, packageName, universalTag) {
+  if (lineText.trim().split(" ")[0] !== "#") {
+    if (lineText.includes(scenario)) {
+      testCountTemplate(testCount, packageName, "totalTest");
+
+      if (universalTag !== undefined) {
+        if (
+          ["broken-test", "pre-condition", "un-verified"].includes(universalTag)
+        ) {
+          universalTag = "other";
+        }
+        testCountTemplate(testCount, packageName, universalTag);
+      }
+    }
+
+    if (lineText.includes(manual)) {
+      testCountTemplate(testCount, packageName, "manual");
+    }
+
+    if (lineText.includes(smoke)) {
+      testCountTemplate(testCount, packageName, "smoke");
+    }
+
+    if (lineText.includes(smoke) && lineText.includes(todo)) {
+      testCountTemplate(testCount, packageName, "smoke_todo");
+    }
+
+    if (lineText.includes(regres)) {
+      testCountTemplate(testCount, packageName, "regres");
+    }
+
+    if (lineText.includes(regres) && lineText.includes(todo)) {
+      testCountTemplate(testCount, packageName, "regres_todo");
+    }
+
+    if (
+      lineText.includes(broken) ||
+      lineText.includes(unverified) ||
+      lineText.includes(precondition)
+    ) {
+      if (lineText.includes(todo)) {
+        testCountTemplate(testCount, packageName, "other_todo");
+      } else {
+        testCountTemplate(testCount, packageName, "other");
+      }
+    }
+  }
+}
+
+function fileParser(fileName, directoryPath) {
+  let universalTag;
+  let packageName;
+  const directoryElements = directoryPath.split("/");
+  testCount.set(`${fileName}_line_no`, 0);
+
+  var r = readline.createInterface({
+    input: fs.createReadStream(directoryPath + "/" + fileName),
+  });
+
+  r.on("line", function (text) {
+    if (testCount.get(`${fileName}_line_no`) === 0) {
+      if (text.includes("@")) {
+        packageName = text.trim().split("@")[1].trim();
+        if (tags.includes(packageName)) {
+          packageName =
+            directoryElements[directoryElements.indexOf("packages") + 1].trim();
+        }
+      } else {
+        packageName =
+          directoryElements[directoryElements.indexOf("packages") + 1].trim();
+      }
+
+      if (packageName === "dev-console") {
+        packageName =
+          directoryElements[directoryElements.indexOf("features") + 1].trim();
+      }
+
+      if (!packages.includes(packageName)) {
+        packages.push(packageName);
+      }
+
+      let filteredArray = text.split("@").filter((word) => {
+        if (tags.includes(word)) return word;
+      }); // it is assumend that only one tag is filtered
+      universalTag = filteredArray[0];
```

Comment:
Why is that assumed? You can use `.find` instead of `.filter` to get the first match.